Modelica/ComplexBlocks/ComplexMath/TransferFunction.mo

```diff
@@ -15,9 +15,16 @@ protected
   Complex aw[size(a,1)];
   Complex bSum;
   Complex aSum;
+  function PowerOfI "Internal function to compute j^k for any integer"
+    input Integer k;
+    output Complex x;
+  algorithm
+   x:=if mod(k,4)==1 then Complex(1,0) elseif mod(k,4)==2 then j elseif mod(k,4)==3 then Complex(-1,0) else -j;
```

Comment:
Maybe use an internal variable to store mod(k,4) and explicitly set the Inline=true annotation.

The second argument of Complex ctor can be skipped.

Reply:
Ok, did that - and corrected an error that didn't influence the end-results as well.